tasks/resolve_reference.rb

```diff
@@ -91,69 +94,38 @@ def extract_resources(state)
     else
       state.fetch('modules', {}).flat_map do |mod|
         mod.fetch('resources', {}).map do |name, resource|
-          [name, resource.dig('primary', 'attributes')]
+          data = resource.dig('primary', 'attributes').reject { |k, _v| k[-1] == '#' }
```

Comment:
This could use a comment I think. Why do we need to reject those?

Reply:
A version 3 state file uses dot notation instead of a nested structure and adds fields that let you reference how many elements are in an array in that structure.

```
nested.data.structure.0.value: "foo";
nested.data.structure.1.value: "bar";
nested.data.structure.#: 2;
```

Leaving in fields that end with `#` causes issues when building the nested data structure, as it would try to set the value at `nested.data.structure` as a hash with key `#`, when the value should be an array.

Reply:
ahh, nice yeah a comment there would be super helpful.